crates/holochain_p2p/src/lib.rs

```diff
@@ -167,6 +166,115 @@ pub trait HolochainP2pDnaT: Send + Sync {
+            &self,
+            from_agent: AgentPubKey,
+            to_agent_list: Vec<(Signature, AgentPubKey)>,
+            zome_name: ZomeName,
+            fn_name: FunctionName,
+            cap: Option<CapSecret>,
+            payload: ExternIO,
+            nonce: Nonce256Bits,
+            expires_at: Timestamp,
+        ) -> actor::HolochainP2pResult<()>;
+        #[allow(clippy::too_many_arguments)]
+        async fn publish(
+            &self,
+            request_validation_receipt: bool,
+            countersigning_session: bool,
+            basis_hash: holo_hash::OpBasis,
+            source: AgentPubKey,
+            op_hash_list: Vec<OpHashSized>,
+            timeout_ms: Option<u64>,
+            reflect_ops: Option<Vec<DhtOp>>,
+        ) -> actor::HolochainP2pResult<()>;
+        async fn publish_countersign(
+            &self,
+            flag: bool,
+            basis_hash: holo_hash::OpBasis,
+            op: DhtOp,
+        ) -> actor::HolochainP2pResult<()>;
+        async fn get(
+            &self,
+            dht_hash: holo_hash::AnyDhtHash,
+            options: actor::GetOptions,
+        ) -> actor::HolochainP2pResult<Vec<WireOps>>;
+        async fn get_meta(
+            &self,
+            dht_hash: holo_hash::AnyDhtHash,
+            options: actor::GetMetaOptions,
+        ) -> actor::HolochainP2pResult<Vec<MetadataSet>>;
+        async fn get_links(
+            &self,
+            link_key: WireLinkKey,
+            options: actor::GetLinksOptions,
+        ) -> actor::HolochainP2pResult<Vec<WireLinkOps>>;
+        async fn count_links(
+            &self,
+            query: WireLinkQuery,
+        ) -> actor::HolochainP2pResult<CountLinksResponse>;
+        async fn get_agent_activity(
+            &self,
+            agent: AgentPubKey,
+            query: ChainQueryFilter,
+            options: actor::GetActivityOptions,
+        ) -> actor::HolochainP2pResult<Vec<AgentActivityResponse<ActionHash>>>;
+        async fn must_get_agent_activity(
+            &self,
+            author: AgentPubKey,
+            filter: holochain_zome_types::chain::ChainFilter,
+        ) -> actor::HolochainP2pResult<Vec<MustGetAgentActivityResponse>>;
+        async fn send_validation_receipts(
+            &self,
+            to_agent: AgentPubKey,
+            receipts: ValidationReceiptBundle,
+        ) -> actor::HolochainP2pResult<()>;
+        async fn authority_for_hash(
+            &self,
+            basis: holo_hash::OpBasis,
+        ) -> actor::HolochainP2pResult<bool>;
+        async fn countersigning_session_negotiation(
+            &self,
+            agents: Vec<AgentPubKey>,
+            message: event::CountersigningSessionNegotiationMessage,
+        ) -> actor::HolochainP2pResult<()>;
+        async fn new_integrated_data(&self) -> actor::HolochainP2pResult<()>;
+        fn chc(&self) -> Option<ChcImpl>;
+    }
+    impl Clone for HolochainP2pDnaT {
+        fn clone(&self) -> Self;
+    }
```

Comment:
Could you just do a real impl for Clone? I don't think you want to test alternate implementations of Clone, you just want the thing cloned, right?

Reply:
I'm not sure how, the MockXXX type is generated. The docs say that `Clone` isn't supported and you need to switch to `mock!`

> For structs with more than one impl block or that have unsupported #[derive(X)] attributes, e.g. Clone, see [mock!](https://docs.rs/mockall/latest/mockall/macro.mock.html) instead.

But if there's a simpler way to do this then yes please because it's a bit untidy as it is

Reply:
@ThetaSinner ah, I didn't see that there are two places you're defining `expect_clone`, one of which actually contains useful info. I thought you were just stubbing it out by creating a new mock, but I see now that the implementation does matter. So disregard this comment.

Yeah, I guess the only correct thing to do is to refactor to put this into an Arc, which you mentioned as a TODO.